internal/dev_server/model/project.go

```diff
@@ -71,7 +74,7 @@ func UpdateProject(ctx context.Context, projectKey string, context *ldcontext.Co
 		return Project{}, err
 	}
 	if !updated {
-		return Project{}, err
+		return Project{}, errors.New("Project not updated")
```

Comment:
For `UpdateProject`, not returning an error here if the project wasn't actually updated seems like a bug? I updated this and `SyncProject` to return an error in this case, as well as coverage for that case. However, I definitely lack a lot of context so lemme know if there's something I'm missing.

Reply:
Oh, yeah. that makes sense. TBH we should probably change the store interface so that we return an error instead of the boolean. Issue is that if no rows are updated when we try and update, it means that we couldn't find the project.